openmc/material.py

```diff
@@ -973,3 +973,73 @@ def export_to_xml(self, path='materials.xml'):
         # Write the XML Tree to the materials.xml file
         tree = ET.ElementTree(root_element)
         tree.write(path, xml_declaration=True, encoding='utf-8')
+
+
+def make_boric_acid(boron_ppm, temperature=293., pressure=0.1013, density=None,
+                    **kwargs):
+    """Return a Material with the composition of boric acid.
+
+    The water density can either be given directly, or it can be determined from
+    a temperature and pressure.
+
+    Parameters
+    ----------
+    boron_ppm : float
+        The weight fraction in parts-per-million of elemental boron in the acid.
+    temperature : float
+        Water temperature in [K] used to compute water density.
+    pressure : float
+        Water pressure in [MPa] used to compute water density.
+    density : float
+        Water density in [g / cm^3].  If specified, this value overrides the
+        temperature and pressure arguments.
+    **kwargs
+        All keyword arguments are passed to the created Material object.
+
+    Returns
+    -------
+    openmc.Material
+
+    """
+    # Set the density of water, either from an explicitly given density or from
+    # temperature and pressure.
+    if density is not None:
+        water_density = density
+    else:
+        water_density = openmc.data.water_density(temperature, pressure)
+
+    # Compute the density of the boric acid.
+    acid_density = water_density / (1 - boron_ppm * 1e-6)
+
+    # Compute the molar mass of pure water.
+    hydrogen = openmc.Element('H')
+    oxygen = openmc.Element('O')
+    M_H2O = 0.0
+    for iso_name, frac, junk in hydrogen.expand(2.0, 'ao'):
+        M_H2O += frac * openmc.data.atomic_mass(iso_name)
+    for iso_name, frac, junk in oxygen.expand(1.0, 'ao'):
+        M_H2O += frac * openmc.data.atomic_mass(iso_name)
+
+    # Compute the molar mass of boron.
+    boron = openmc.Element('B')
+    M_B = 0.0
+    for iso_name, frac, junk in boron.expand(1.0, 'ao'):
+        M_B += frac * openmc.data.atomic_mass(iso_name)
+
+    # Compute the number fractions of each element.
+    frac_H2O = (1 - boron_ppm * 1e-6) / M_H2O
+    frac_H = 2 * frac_H2O
+    frac_O = frac_H2O
+    frac_B = boron_ppm * 1e-6 / M_B
```

Comment:
This computation neglects the H and O from the boric acid, while i would agree that should be a negligible impact, is that commonplace?  If so, for clarity's sake, should it be noted?

Reply:
I suspect it's commonplace, but I'm not 100% sure.  The BEAVRS benchmark does it this way.  Looking at the atom densities in BAW-1484-7, they maintain a 2-1 H-O ratio even at 1000 ppm.

I thought about maybe including the H and O from the boric acid just to be super rigorous, but that looks like it can be surprisingly complicated.  Wikipedia tells me that at 170 and 236 deg C, boric acid partly breaks down producing water, so we would maybe have to incorporate that into the density calculation.  And as you say, the impact is almost certainly insignificant.

Taking your suggested change of calling it "borated water" nicely makes it more clear how we did the calculation.  I also added a bit to the docstring to be really explicit.